core/src/com/unciv/logic/automation/unit/WorkerAutomation.kt

```diff
@@ -279,6 +279,13 @@ class WorkerAutomation(
                 }
                 .sortedByDescending { getPriority(it) }
 
+        // Carthage can move through mountains, special case
+        val mountainWorkableTiles = workableTiles.filter{tile -> tile.isMountain()}
```

Comment:
I feel like this can be made more generic to include modded terrains that deals damage. Perhaps check if the tile has the unique to damage units?

Reply:
Haven't thought about that, now it should take that into consideration.